### DISCUSSION TITLE
Integration with Rage

### DISCUSSION THREAD

**Discussion by rsamoilov** [okuramasafumi/alba#400] [General]
Hey @okuramasafumi,

I’m the maintainer of the [Rage](https://github.com/rage-rb/rage) framework. We’ve recently released a feature that enables building OpenAPI specifications for Rage apps, and one of its features is [integration with Alba](https://github.com/rage-rb/rage/wiki/OpenAPI-Guide#responses).

If you’re ever interested in merging this code into Alba, please let me know. Otherwise, I just wanted to share that this integration now exists and to thank you for creating such an amazing gem!

Cheers!

**Comment by okuramasafumi** [comment]
@rsamoilov Hi, thank you for adding Alba integration to Rage! I know Rage since it appeared on Ruby Weekly and I'm interested in using while I have not had a chance to.

Yes, I'm interested in having some part of the code in Alba itself. I'm also willing to add things you felt you miss so that future integrations will be easier.

**Reply by rsamoilov** [reply]
OK, now I feel self-conscious because the code is really ugly - [AlbaParser](https://github.com/rage-rb/rage/blob/master/lib/rage/openapi/parsers/ext/alba.rb) 😄 
From what I see, only [layouts](https://github.com/okuramasafumi/alba?tab=readme-ov-file#layout) and [helpers](https://github.com/okuramasafumi/alba?tab=readme-ov-file#helper) are currently not supported. Everything else should work just fine.

Another known issue is the `collection_key` call inside inline associations.

I’ll be slightly updating the code in the next weeks, but let me know how you want to proceed!

**Reply by okuramasafumi** [reply]
You use Prism to parse Alba, which is beyond my expectation😁
So I consider if we can replace a parser with more robust, official API.

For example, if Alba has `analyze` method (this name is temporary), and we have a resource class, we can do something like:

```ruby
content = File.read(path_to_resource)
eval(content + '; class_name.analyze')
```

What do you think about it?

**Reply by rsamoilov** [reply]
Prism actually works great here. But in general, I felt it would be more productive to start a conversation with something working rather than with a sketch of an idea.

And yes, an official API that allows different tools to build an OpenAPI specification for a resource looks like a perfect solution to me!

**Comment by rsamoilov** [comment]
Hey @okuramasafumi 

If you're interested, here's a guide on building APIs with OpenAPI documentation using Rage and Alba.
https://zuplo.com/blog/2025/04/13/ruby-rage-rest-api-tutorial
